django/db/models/expressions.py

```diff
@@ -542,16 +542,24 @@ def as_sql(self, compiler, connection, function=None, template=None):
             arg_sql, arg_params = compiler.compile(arg)
             sql_parts.append(arg_sql)
             params.extend(arg_params)
-        if function is None:
-            self.extra['function'] = self.extra.get('function', self.function)
-        else:
-            self.extra['function'] = function
-        self.extra['expressions'] = self.extra['field'] = self.arg_joiner.join(sql_parts)
-        template = template or self.extra.get('template', self.template)
-        return template % self.extra, params
-
-    def as_sqlite(self, *args, **kwargs):
-        sql, params = self.as_sql(*args, **kwargs)
+        data = self.extra.copy()
+        data.update(**extra_context)
+        # we add to the data dict because these values could have been supplied
+        # in self.extra also
+        if function is not None:
+            data['function'] = function
+        if template is not None:
+            data['template'] = template
```

Comment:
I think there's no reason that 'template' and 'arg_joiner' need to go into `data` -- they aren't meant to be interpolated into `template`, correct? Instead of the `data.get()` lines below, wouldn't this work:
```python
template = template or self.template
arg_joiner = arg_joiner or self.arg_joiner
```